lib/ruby_lsp/requests/support/semantic_token_encoder.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+module RubyLsp
+  module Requests
+    module Support
+      class SemanticTokenEncoder
+        def initialize
+          @current_row = 0
+          @current_column = 0
+        end
+
+        def encode(tokens)
+          tokens = sort(tokens)
+
+          delta = tokens.flat_map do |token|
+            compute_delta(token)
+          end
```

Comment:
I think we can easily change this to:
```suggestion
          delta = tokens
            .sort_by do |token|
              [token.location,start_line, token.location.start_column]
            end
            .flat_map do |token|
              compute_delta(token)
            end
```
and drop the `sort` method.

Btw, the sorting order in the `sort` method is not correct, right? It seems to be sorting by column first and row last, but it should be the other way around.